### PR TITLE
docs(getting started): Change dataset to synthetic

### LINKED ISSUE #1861: docs: Change the data used in the getting started guide
### Which part of the documentation needs improvement?

Following file: `examples/getting_started/plot_skore_getting_started.py`

### Describe the problem found in the documentation

The following dataset is used:
```python
from sklearn.datasets import load_breast_cancer

X, y = load_breast_cancer(return_X_y=True, as_frame=True)
```

This is a real-world dataset but it's too simple so the results look a bit too nice and it's hard to make a very compelling example usage of skore with this dataset:

Image

### Suggested improvement

Try a more difficult dataset, a synthetic but hard one, as done in the user guide at `sphinx/user_guide/displays.rst`:

```python
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression
from skore import CrossValidationReport

X, y = make_classification(
    n_samples=10_000,
    n_classes=3,
    class_sep=0.3,
    n_clusters_per_class=1,
    random_state=42,
)
report = CrossValidationReport(LogisticRegression(), X, y)
disp = report.metrics.roc()
disp.plot()
```
which renders:

Image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mohsinm-dev**
## Summary
  - Replace `load_breast_cancer` with `make_classification` using challenging parameters
  - Creates a more compelling demonstration of Skore's capabilities
  - The breast cancer dataset produces near-perfect results that don't effectively showcase the model
  evaluation tools

  ## Changes
  - Use `make_classification` with parameters: n_samples=1000, n_features=20, n_informative=10, 
  n_redundant=10, n_clusters_per_class=1, random_state=42
  - Convert to DataFrame format to maintain compatibility with existing code
  - Update comments to reflect synthetic dataset usage

  ## Why this improves the example
  The synthetic dataset provides more realistic performance metrics (ROC-AUC ~0.90 vs ~0.99) and
  clearer differences between models, making it better for educational purposes, and demonstrating
  When different models/techniques matter.

  ## Test plan
  - [x] Verified EstimatorReport works with the synthetic dataset
  - [x] Tested CrossValidationReport functionality
  - [x] Confirmed ComparisonReport works correctly
  - [x] Validated feature importance calculations
  - [x] Ensured all metrics and visualizations display properly

  Fixes #1861

### CODE REVIEW

File: examples/getting_started/plot_skore_getting_started.py [hunk 1]
```diff
@@ -40,17 +40,29 @@
...
+    n_features=20,
+    n_informative=10,
+    n_redundant=10,
+    n_clusters_per_class=1,
+    random_state=42,
+)
+
+# Convert to DataFrame format to match the original example structure
+X = pd.DataFrame(X, columns=[f"feature_{i}" for i in range(X.shape[1])])
+y = pd.Series(y, name="target")
```
Comment (thomass-dev) [inline examples/getting_started/plot_skore_getting_started.py]: `X` and `y` already have the good format. Please remove this section.

**Comment by thomass-dev** [comment]
Did you use an LLM to generate your PR? Please check the [documentation](https://github.com/probabl-ai/skore/blob/main/CONTRIBUTING.rst#automated-contributions-policy).

The challenge is to understand the issue, and to be able to easily contribute in future.
Thanks.

**Comment by mohsinm-dev** [comment]
> Did you use an LLM to generate your PR? Please check the [documentation](https://github.com/probabl-ai/skore/blob/main/CONTRIBUTING.rst#automated-contributions-policy).
> 
> The challenge is to understand the issue, and to be able to easily contribute in the future. Thanks.


Not really, I did take some help from LLM to understand the issue better, but did not generate the PR using LLM. Let me know if there is any issues, I'll look into it and submit the PR again, Thanks.

File: examples/getting_started/plot_skore_getting_started.py (hunk 1 above)
Comment (mohsinm-dev) [reply examples/getting_started/plot_skore_getting_started.py]: I was thinking that since the original `load_breast_cancer(return_X_y=True, as_frame=True)` returned pandas DataFrames, I needed to convert the make_classification output to maintain the same data types. But after testing, I realized that skore works perfectly fine with numpy arrays directly, so the conversion was unnecessary. I've removed the DataFrame conversion code and signed the commit.

File: examples/getting_started/plot_skore_getting_started.py [hunk 2]
```diff
@@ -227,7 +236,6 @@
 # Let us load a dataset containing some time series data:
 
 # %%
-import pandas as pd
```
Comment (thomass-dev) [inline examples/getting_started/plot_skore_getting_started.py]: Please can you revert this unnecessary change?

**Review by thomass-dev — CHANGES_REQUESTED** [review]
I don't think the PR is sufficient to resolve the issue. @sylvaincom can you take a look? Thanks!

This is the current rendering:
image

**Comment by sylvaincom** [comment]
Hi, I agree with @thomass-dev: this feels LLM generated and you did not take into account the description of the parent issue, especially the shown plots

`comparator.metrics.roc().plot()` should display a ROC curve that does not appear too easy

**Comment by mohsinm-dev** [comment]
@sylvaincom  I’ve updated the example to use 10,000 samples with `CrossValidationReport `and matched the code structure to your reference. AUC values may vary slightly due to dataset generation, but the code, setup, and plot structure are identical. Let me know if you’d prefer we lock a seed or dataset for consistency.

Screenshot 2025-07-29 at 3 45 13 PM

**Comment by mohsinm-dev** [comment]
@sylvaincom  I’ve updated the example to use 10,000 samples with `CrossValidationReport` and matched the code structure to your reference. Can you please check this ?

Screenshot 2025-07-29 at 4 13 00 PM

**Comment by sylvaincom** [comment]
You do not seem to understand what you're doing: you're calling `rf_report` (RF : random forest) a model report corresponding to a logistic regression

File: examples/getting_started/plot_skore_getting_started.py [hunk 3]
```diff
@@ -40,31 +40,37 @@
 # In order to assist its users when programming, skore has implemented a
 # :class:`skore.EstimatorReport` class.
 #
-# Let us load a binary classification dataset and get the estimator report for a
-# :class:`~sklearn.ensemble.RandomForestClassifier`:
+# Let us load a multi-class classification dataset and get the estimator report for a
```
Comment (glemaitre) [inline examples/getting_started/plot_skore_getting_started.py]: By using `make_classification`, we are not loading a dataset anymore but generating one. For a getting started, I would say that it could be good enough but we need to change the sentence then.

File: examples/getting_started/plot_skore_getting_started.py [hunk 4]
```diff
@@ -40,31 +40,37 @@
...
+# :class:`~sklearn.linear_model.LogisticRegression`:
 
 # %%
-from sklearn.datasets import load_breast_cancer
-from sklearn.ensemble import RandomForestClassifier
+from sklearn.datasets import make_classification
+from sklearn.linear_model import LogisticRegression
```
Comment (glemaitre) [inline examples/getting_started/plot_skore_getting_started.py]: What is the reason to have change the `RandomForestClassifier` by the `LogisticRegression` model?

Comment (glemaitre) [reply examples/getting_started/plot_skore_getting_started.py]: In any case, we need to regenerate the static figure at the end of the document that is located in `_static/images/screenshot_getting_started.png`

File: examples/getting_started/plot_skore_getting_started.py [hunk 5]
```diff
@@ -40,31 +40,37 @@
...
+    n_samples=10_000,
+    n_classes=3,
+    class_sep=0.3,
+    n_clusters_per_class=1,
+    random_state=42,
+)
 X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
 
-rf = RandomForestClassifier(random_state=0)
+lr = LogisticRegression(random_state=0)
```
Comment (glemaitre) [inline examples/getting_started/plot_skore_getting_started.py]: If we go with a `LogisticRegression`, we don't need to set the `random_state` because it uses LBFGS that is deterministic.

File: examples/getting_started/plot_skore_getting_started.py (hunk 4 above)
Comment (mohsinm-dev) [reply examples/getting_started/plot_skore_getting_started.py]: @glemaitre Thanks for the feedback. I’ve replicated the changes with the improvements mentioned in #1861. The main purpose was to test the synthetic data as outlined in the user guide (sphinx/user_guide/displays.rst), so I agree that switching to LogisticRegression wasn’t necessary. Let me know if we should keep LogisticRegression or switch to RandomForest for consistency or further testing.

Comment (auguste-probabl) [reply examples/getting_started/plot_skore_getting_started.py]: reverted to RandomForest

**Comment by auguste-probabl** [comment]
Here's what you get with RandomForestClassifier:

CVReport:
2025-08-28T10_18_22_screenshot

ComparisonReport:
2025-08-28T10_19_20_screenshot

IMO it looks good enough for our purposes.

**Review by thomass-dev — APPROVED** [review]
Thanks @auguste-probabl for the follow-up, LGTM now.
